OpenRA.Mods.Common/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -151,7 +151,8 @@ public override bool HandleMouseInput(MouseInput mi)
 						// World.CancelInputMode. If we did check it, actor de-selection would not be possible by just clicking somewhere,
 						// only by dragging an empty selection box.
 						*/
-						if (isDragging && (!(World.OrderGenerator is GenericSelectTarget) || IsValidDragbox))
+						var uog = (UnitOrderGenerator)World.OrderGenerator;
```

Comment:
We check `World.OrderGenerator is UnitOrderGenerator` above, so imho that should be replaced by using `as` and checking for `null` there and dropping the cast here.

Reply:
Its actually even worse than that - the check you refer to is redundant, because there is an even earlier check that routes everything that isn't UOG down a different code path.